tools/jenkins/Jenkinsfile

```
@@ -17,38 +17,33 @@
  */
 
 timeout(time: 4, unit: 'HOURS') {
-
-    node("openwhisk2") {
+    node("openwhisk1") {
         deleteDir()
-        stage ('Checkout and build on OpenWhisk2') {
+        stage ('Checkout') {
             checkout([$class: 'GitSCM', branches: [[name: '${Branch}']], doGenerateSubmoduleConfigurations: false,
                       extensions: [], submoduleCfg: [],
                       userRemoteConfigs: [[credentialsId: '', url: 'https://github.com/${Fork}/${RepoName}']]])
-            sh './gradlew :core:invoker:distDocker'
         }
-    }
-
-    node("openwhisk3") {
-        deleteDir()
-        stage ('Checkout and build on OpenWhisk3') {
-            checkout([$class: 'GitSCM', branches: [[name: '${Branch}']], doGenerateSubmoduleConfigurations: false,
-                     extensions: [], submoduleCfg: [],
-                     userRemoteConfigs: [[credentialsId: '', url: 'https://github.com/${Fork}/${RepoName}']]])
-            sh './gradlew :core:invoker:distDocker'
-       }
-    }
 
-    node("openwhisk1") {
-        deleteDir()
-        stage ('Checkout and build on OpenWhisk1') {
-            checkout([$class: 'GitSCM', branches: [[name: '${Branch}']], doGenerateSubmoduleConfigurations: false,
-                      extensions: [], submoduleCfg: [],
-                      userRemoteConfigs: [[credentialsId: '', url: 'https://github.com/${Fork}/${RepoName}']]])
+        stage ('Build') {
+            // Set up a private docker registry service, accessed by all the OpenWhisk VMs.
+            sh 'docker container stop registry && docker container rm -v registry'
+            sh 'docker run -d --restart=always --name registry -v "$HOME"/certs:/certs \
+                -e REGISTRY_HTTP_ADDR=0.0.0.0:444 -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+                -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key -p 444:444 registry:2'
+            // Build the controller and invoker images.
             sh './gradlew distDocker'
```

Instead of tag and push individually, gradle script accepts dockerRegistry as a parameter. We can use below

```suggestion
            sh './gradlew distDocker -PdockerRegistry=openwhisk-vm1-he-de.apache.org:444'
```

---

Done. Thx.